spark3/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -238,9 +247,8 @@ public StreamingOffset initialOffset() {
       }
 
       table.refresh();
-      StreamingOffset offset = table.currentSnapshot() == null ?
-          StreamingOffset.START_OFFSET :
-          new StreamingOffset(SnapshotUtil.oldestSnapshot(table).snapshotId(), 0, false);
+      StreamingOffset offset = isStreamEmpty(table, fromTimestamp) ? StreamingOffset.START_OFFSET :
+          new StreamingOffset(SnapshotUtil.oldestSnapshot(table, fromTimestamp).snapshotId(), 0, false);
```

Comment:
I don't think this is correct. If there is no current snapshot, then `START_OFFSET` is correct. But if there is a current snapshot and the starting timestamp is after that snapshot, then the offset should be set as though current was completely processed. `START_OFFSET` is not appropriate in that case because that requires resuming from a snapshot with no parent.

Reply:
Not sure I understanding the second question correctly. Let's take an example
Let's say we have snapshots 5, 10, 15, 20, 25
Current snapshot is 25
fromTimestamp is 30
In such case isStreamEmpty will return false, offset will be -1
so we will read everything in the stream 5, 10, 15, 20, 25. I think this is the expected behavior. @SreeramGarlapati what do you think?

Reply:
That doesn't sound like the intended behavior - or at least the expected behavior to me.

I would expect that if currentSnapshot is 25 and we request fromTimestamp of 30, the stream would not read anything. It would be assumed the user does not want any new updates as they wanted all changed data after 30, for which there is none.

Reply:
Sorry my understanding is wrong. @kbendick Yes, that's exactly the behavior.
Just create a unit test to prove the above scenario. When we pass 30, the stream processing will return nothing
Because:
`  @Override
  public InputPartition[] planInputPartitions(Offset start, Offset end) {
    Preconditions.checkArgument(end instanceof StreamingOffset, "Invalid end offset: %s is not a StreamingOffset", end);
    Preconditions.checkArgument(
        start instanceof StreamingOffset, "Invalid start offset: %s is not a StreamingOffset", start);

    if (end.equals(StreamingOffset.START_OFFSET)) {
      return new InputPartition[0];
    }`

In this case start == end, we won't have anything offset to process